ww

```diff
@@ -78,86 +94,122 @@ EOF
     if [[ "$kwinMajorVersion" -lt "$minimumVersion" ]]; then
-        echo "ERROR: This feature needs KWin $minimumVersion or later." >&2 
+        echo "ERROR: This feature needs KWin $minimumVersion or later." >&2
         exit 1
     fi
 
     # This way is similar to the one used on https://discuss.kde.org/t/xdotool-replacement-on-wayland/7242/9
     jsFile="$(mktemp)" || exit 1   # It is the file where the javascript code is going to be saved
     echo "print(\"$jsFile\",workspace.activeWindow.internalId);" > "$jsFile" || exit 1
-    
-    scriptId="$(qdbus org.kde.KWin /Scripting loadScript "$jsFile")" || exit 1
+
+    if [ "$USE_DBUS_SEND" = true ]; then
+        scriptId="$(dbus-send --session --dest=org.kde.KWin --print-reply=literal /Scripting org.kde.kwin.Scripting.loadScript string:"$jsFile" string:"info_script" | awk '{print $2}')" || exit 1
+    else
+        scriptId="$(qdbus org.kde.KWin /Scripting loadScript "$jsFile")" || exit 1
+    fi
+
     timestamp="$(date +"%Y-%m-%d %H:%M:%S")" || exit 1
-    # Starts the script
-    qdbus org.kde.KWin /Scripting/Script"$scriptId" run || exit 1
     
+    # Starts the script
+    if [ "$USE_DBUS_SEND" = true ]; then
+        dbus-send --session --dest=org.kde.KWin --print-reply=literal /Scripting/Script"$scriptId" org.kde.kwin.Script.run || exit 1
+    else
+        qdbus org.kde.KWin /Scripting/Script"$scriptId" run || exit 1
+    fi
+
     # Uses some arguments that are also seen on https://github.com/jinliu/kdotool/blob/master/src/main.rs
     outputJournalctl="$(journalctl --since "$timestamp" --user --user-unit=plasma-kwin_wayland.service --user-unit=plasma-kwin_x11.service --output=cat -g "js: $jsFile")" || exit 1
     # Uses `awk` separately in order to avoid masking a return value, as Shellcheck recommends
     windowId="$(awk '{print $3}' <<< "$outputJournalctl")" || exit 1
-    # Stops the script
-    qdbus org.kde.KWin /Scripting/Script"$scriptId" stop || exit 1
     
+    # Stops the script
+    if [ "$USE_DBUS_SEND" = true ]; then
+        dbus-send --session --dest=org.kde.KWin --print-reply=literal /Scripting/Script"$scriptId" org.kde.kwin.Script.stop || exit 1
+    else
+        qdbus org.kde.KWin /Scripting/Script"$scriptId" stop || exit 1
+    fi
+
     # Shows the information about that window
-    qdbus org.kde.KWin /KWin org.kde.KWin.getWindowInfo "$windowId" || exit 1
-    
+    if [ "$USE_DBUS_SEND" = true ]; then
+        dbus-send --session --dest=org.kde.KWin --print-reply=literal /KWin org.kde.KWin.getWindowInfo uint32:"$windowId" || exit 1
+    else
+        qdbus org.kde.KWin /KWin org.kde.KWin.getWindowInfo "$windowId" || exit 1
+    fi
+
     exit 0
 fi
 
 SCRIPT_TEMPLATE=$(
 	cat <<EOF
 function kwinactivateclient(clientClass, clientCaption, toggle) {
-    var clients = workspace.clientList ? workspace.clientList() : workspace.windowList();
-    var activeWindow = workspace.activeClient || workspace.activeWindow;
+    var clients = workspace.windowList();
```

Comment:
do not change this script. it is not part of this PR and it is breaking compatibility with other kde versions